python/lsst/daf/butler/core/datastore.py

```diff
@@ -638,6 +638,43 @@ def getURI(self, datasetRef: DatasetRef, predict: bool = False) -> ButlerURI:
         """
         raise NotImplementedError("Must be implemented by subclass")
 
+    @abstractmethod
+    def getArtifacts(self, refs: Iterable[DatasetRef],
```

Comment:
I think I'd prefer something like "move" or "retrieve" than "get" here; typically we use "get" to return Python objects.

Reply:
Okay. I'll use `retrieve` then. I wasn't sure how you wanted to play it (we do have `getURI`). I didn't want to use `copy` because we don't copy sometimes, we link.